lib/config.js

```diff
@@ -95,64 +95,16 @@ if (options.enableTracing) process.env.STRONGLOOP_TRACING = 1;
 // Communicate detach option to the master using config.
 config.detach = options.detach;
 
-var app = process.argv[2];
-var appRoot;
-
-try {
-  var stat = fs.statSync(app);
-} catch (er) {
-  try {
-    app = require.resolve(path.resolve(app));
-    stat = fs.statSync(app);
-  } catch (er) {
-    console.error('Invalid app (%s), try `%s --help`.\n', er, options.NAME);
-    process.exit(1);
-  }
-}
-
-// We want to be in the directory of the file we are running, so we can pick up
-// configuration stored in it's working directory.
-
-if (stat.isFile()) {
-  // The app is a file, we'll run it from its module's root directory.
-
-  // XXX(sam) this branch causes PWD to be set to CWD, fix might be:
-  //   app = path.resolve(process.env.PWD, app)
-  // which should perhaps be done before the statSync above
-  app = path.resolve(app);
-  appRoot = path.dirname(app);
-  app = path.basename(app);
-  // walk up the path until we find the package.json nearest to the app
-  while (!fs.existsSync(path.join(appRoot, 'package.json'))) {
-    app = path.join(path.basename(appRoot), app);
-    // hit root, fallback to running as-is
-    if (appRoot === path.dirname(appRoot)) {
-      app = path.basename(process.argv[2]);
-      appRoot = path.dirname(process.argv[2]);
-      break;
-    }
-    appRoot = path.dirname(appRoot);
-  }
-  chdir(appRoot);
-}
-
-if (stat.isDirectory()) {
-  // The app is a directory (such as '.'), we'll look for what to run from it.
-  chdir(app);
-
-  // The canonical way to run is with `npm start`, but we can't do that for a
-  // cluster worker, because the worker must start the runner (so it can require
-  // strong-agent). Instead, we search for a runnable file, and if that fails,
-  // just require the directory and let node do the right thing (which should be
-  // to use package.main if it exists, index.js otherwise).
-  function existing(file) {
-    return fs.existsSync(file) ? file : null;
-  }
-  app = existing('server.js') || existing('app.js') || '.';
+var app = startCmd(process.cwd(), process.argv);
+if (app.error) {
+  console.error('Invalid app (%s), try `%s --help`.\n',
+                app.error, options.NAME);
```

Comment:
should be app.error.message. app.error is an Error object, as-is, this will be `Invalid app (Error: <message>) ...`, but would be better to not have the class name in the message.